src/Sentry/Internal/TcpPing.cs

```diff
@@ -0,0 +1,25 @@
+namespace Sentry.Internal;
+
+internal interface IPing
+{
+    Task<bool> IsAvailableAsync();
+}
+
+internal class TcpPing(string hostToCheck, int portToCheck = 443) : IPing
+{
+    private readonly Ping _ping = new();
+
+    public async Task<bool> IsAvailableAsync()
+    {
+        try
+        {
+            using var tcpClient = new TcpClient();
```

Comment:
I take it the new TcpClient every 500ms is not an issue?

Reply:
It won't be every 500ms due to the progressive backoff. There would be a first attempt, a subsequent attempts after 500ms, then 1000ms then 2000ms etc.

We could try to use the same TcpClient but it would need to be reset to be usable again after each unsuccessful connection attempt. I could do some benchmarks to check those two different approaches. It's worth noting that there will only be one `PollingNetworkStatusListener` instance in the application though, so I don't think this is really a hot path.